tools/create_testgrid_tickets.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+# This script gets a list of the filed tests from testgrid
+# For each failure, which does not already has an open triaging Jira ticket, it creates one
+#
+
+import argparse
+import logging
+import netrc
+import os
+import sys
+import re
+from urllib.parse import urlparse
+import requests
+import jira
+
+
+DEFAULT_WATCHERS = ["ronniela", "odepaz"]
+
+
+TEST_VERSION_REGEX = re.compile(r'.*-(\d\.[^-]+)-.*')
+TEST_STATUS_URL = "https://testgrid.k8s.io/redhat-assisted-installer/summary"
+TEST_RUNS_URL = 'https://testgrid.k8s.io/redhat-assisted-installer/table?tab={}&width=10&dashboard=redhat-assisted-installer'
+JIRA_SERVER = "https://issues.redhat.com/"
+DEFAULT_NETRC_FILE = "~/.netrc"
+JIRA_SUMMARY = "CI test: {test_id} ({last_pass})"
+
+JIRA_DESCRIPTION = """
+{{color:red}}Do not manually edit this description, it will get automatically over-written{{color}}
+
+*Test Status*: {status}
+*Test Details:* [{test_id}|https://testgrid.k8s.io/redhat-assisted-installer#{test_id}]
+*Last Test failure:* https://prow.ci.openshift.org/view/gcs/origin-ci-test/logs/{test_id}/{last_fail}
+{tests_list}
+"""
+
+
+def get_credentials_from_netrc(server, netrc_file=DEFAULT_NETRC_FILE):
```

Comment:
maybe we should have it in some utility module

Reply:
You are right.
This is a refactor that needs to be done on the entire repo. There are many tools that duplicate this code.
This will be handled as part of https://issues.redhat.com/browse/MGMT-3945